### PR TITLE
:book: Raising event doc page created

### PR DESCRIPTION
**Pull request by Sajiyah-Salat**
Description: Add the documentation to help out users know how to implement and raise events.
NOTE: we had this document in v1 docs but it is outdated and we are shaping it accordingly.
This is https://github.com/kubernetes-sigs/kubebuilder/pull/3375 original pr but during squashing commits I messed up the original file and this is backup.

### CODE REVIEW

**Comment by Sajiyah-Salat** [comment]
Looks like perfect to merge. WDYT?

File: docs/book/src/reference/raising-events.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		// Note that we added the following line: 
+		Recorder: mgr.GetEventRecorderFor("mykind-controller"),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "MyKind")
+		os.Exit(1)
+	}
+```
+
+### Allowing usage of EventRecorder on the Controller
+To raise an event, you must have access to `record.EventRecorder` in the Controller.  Therefore, firstly let's update the controller implementation:
```
Comment (camilamacedo86) [inline docs/book/src/reference/raising-events.md:63]: Nit: space between title and text.

File: docs/book/src/reference/raising-events.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+)
+// MyKindReconciler reconciles a MyKind object
+type MyKindReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	// See that we added the following code to allow us to pass the record.EventRecorder 
+	Recorder record.EventRecorder
+}
+### Passing the EventRecorder to the Controller
+Events are published from a Controller using an [EventRecorder]`type CorrelatorOptions struct`,
```
Comment (camilamacedo86) [inline docs/book/src/reference/raising-events.md:78]: Same

File: docs/book/src/reference/raising-events.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			cr.Name,
+			cr.Namespace))
+```
+
+</aside>
+
+## How to be able to raise Events?
+
+Following are the steps with examples to help you raise events in your controller's reconciliations. 
+Events are published from a Controller using an [EventRecorder]`type CorrelatorOptions struct`,
```
Comment (camilamacedo86) [inline docs/book/src/reference/raising-events.md:47]: It is missing the link:

Screenshot 2023-06-13 at 06 34 19

**Review by camilamacedo86 — APPROVED** [review]
@Sajiyah-Salat,

See that it has a few nits. 
Since they are not a blocker and you are working on this one for a long period I will 

/approved
/lgtm 

But if you have a chance please address them in a follow up.

**Comment by Sajiyah-Salat** [comment]
sorry for issues I will make changes as soon as possible

File: docs/book/src/reference/raising-events.md (hunk 3 above)
Comment (Sajiyah-Salat) [reply docs/book/src/reference/raising-events.md:47]: Should I add [this](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events) or any other link

**Pull request merged**